### ISSUE TITLE
Pretraining BERT without next sentence prediction

### ISSUE THREAD

**Issue by Guitaricet** [google-research/bert#178]
Hi,
I want to pretrain BERT from multilingual checkpoint, but my data is tweets, so next sentence prediction is not a valid task. Is there any way to turn off next sentence prediction subtask? Eg to zero loss for it.

**Comment by Guitaricet** [comment]
`create_pretraining_data.py` expects various documents separated by a new line. I suppose it would be sufficient to just add '\n' after each tweet.

**Comment by xwzhong** [comment]
1. if you just want to train masked language model, just change code here: https://github.com/google-research/bert/blob/master/run_pretraining.py#L148
2. if you just add '\n' after each tweet, the whole file is just one document, it is confilct with the original design.

**Comment by dvector89** [comment]
@xwzhong The original training data may also need to be changed.

**Comment by loveJasmine** [comment]
are you success on it, and branch to share

**Comment by loveJasmine** [comment]
@dvector89 @xwzhong 
how to change 'The original training data ' in your design

**Comment by xwzhong** [comment]
@loveJasmine I did't make a pretrain model without next sentence prediction, but I changed the code for my own experiment, you should modify the create_pretraining_data.py and run_pretraining.py file, mainly delete the code about next sentence prediction.

**Comment by loveJasmine** [comment]
@xwzhong 
any details on how to modify'create_pretraining_data.py ' ?

**Comment by loveJasmine** [comment]
beside modify'create_pretraining_data.py ' to change the data input , and the 'run_pretraining.py#L148' to modify the loss, 
I think maybe we should delete the segment embedding if we only use mask LM

**Comment by xwzhong** [comment]
@loveJasmine mainly delete the code for create next sentence prediction task, what's more, re-write the function for loading data in https://github.com/google-research/bert/blob/master/run_pretraining.py#L331, and delete the code for next sentence task.
yes, the segment embedding is not necessary, you can delete it, I think there will be no harm if you keep it, because all sentence has the same segment embedding.

**Comment by loveJasmine** [comment]
also, if we do not use ' next sentence prediction' on pre-trining, 
does it possible reduce the model comlexity like use smaller layer, smaller hidden , anyone have some tips to share?

**Comment by aditya-malte** [comment]
Hi @Guitaricet ,
Were you successful.
If yes, could you share the modified code for pre training 
Thanks
